payment_paypal/indico_payment_paypal/controllers.py

```diff
@@ -84,14 +88,22 @@ def _verify_business(self):
         return False
 
     def _verify_amount(self):
-        expected_amount = self.registration.price
+        paypal_fixed_fee = float(current_plugin.event_settings.get(self.registration.registration_form.event, 'paypal_fixed_fee'))
+        paypal_percent_fee = float(re.findall(r'^([0-9]+(\.[0-9]+)?)(%)?', current_plugin.event_settings.get(self.registration.registration_form.event, 'paypal_percent_fee'))[0][0])
```

Comment:
regex is not really the most appropriate way to parse numbers. and ideally this should be validated when being input into the settings..

Reply:
not needed anymore, done.